src/app-layout/runtime-drawer/use-runtime-drawer-context.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { RefObject, useCallback, useEffect, useState } from 'react';
+
+import { awsuiPluginsInternal } from '../../internal/plugins/api';
+import { DrawerConfig } from '../../internal/plugins/controllers/drawers';
+
+export const useRuntimeDrawerContext = ({ rootRef }: { rootRef: RefObject<HTMLElement> }) => {
+  const [drawerContext, setDrawerContext] = useState<DrawerConfig | null>(null);
+
+  const getRuntimeDrawerId = useCallback((element: HTMLElement | null): string | null => {
+    if (!element) {
+      return null;
+    }
+
+    const drawerId = (element.parentNode as HTMLElement)?.dataset?.awsuiRuntimeDrawerRootId;
+
+    if (!drawerId) {
+      return getRuntimeDrawerId(element.parentElement);
```

Comment:
Oops, recursion here may cause call stack overflow, if the DOM tree is too large 

Can we use our `findUpUntil` standard util?